exercises/resistor-color-duo/test/Tests.hs

```diff
@@ -1,19 +1,32 @@
 {-# OPTIONS_GHC -fno-warn-type-defaults #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
 {-# LANGUAGE RecordWildCards #-}
 
-import Data.Foldable     (for_)
-import Test.Hspec        (Spec, describe, it, shouldBe)
-import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+import Data.Foldable              (for_)
+import Test.Hspec                 (Spec, describe, context, it, shouldBe)
+import Test.Hspec.Runner          (configFastFail, defaultConfig, hspecWith)
+import Test.QuickCheck            (property)
+import Test.QuickCheck.Gen        (oneof)
+import Test.QuickCheck.Arbitrary
 
 import ResistorColors (Color(..), value)
 
+instance Arbitrary Color where
+  arbitrary = oneof $ map return [Black, Brown, Red, Orange, Yellow, Green, Blue, Violet, Grey, White]
+
 main :: IO ()
 main = hspecWith defaultConfig {configFastFail = True} specs
 
 specs :: Spec
-specs = describe "value" $ for_ cases test
-  where
+specs = describe "value" $ do
+  context "equality tests" $ for_ cases test
 
+  context "property tests" $
+    it "all values starting with Black are single digit" $ property $
+      \color -> value (Black, color) < 10
+    -- Add more property tests here
```

Comment:
Follow-up comment: Perhaps it is also a good student exercise to write the shrink function.

So you might write

```haskell
-- This could be improved 
shrinkColor :: Color -> [Color]
shrinkColor Black = []
shrinkColor Brown = [Black]
shrinkColor color = [Black, Brown]
```

although I feel like this is deviating into a style of submitting something to the student that appears half-done, which is not a style we've used previously, to my knowledge, on this track. So perhaps it is better to either omit shrinking entirely and use `forAll` rather than `forAllShrink`, and simply hint that shrinking is possible (or not; the articles being linked to elaborate on shrinking, and a lot of students don't follow those links, which is fine).